### PR TITLE
Mike - E11-B1: Create rec req table

### LINKED ISSUE #30: Epic 11 - B1: Create Recommendation Request Table
Note: Please read through all the instructions on this epic - we do not expect you to complete all of it at once, and you should ABSOLUTELY split it into separate issues and PRs. Feel free to tackle as much as you can, in order as specified below. Both epics in this application are important to set the stage for the rest of the features.

### Have you updated the title?

- [X] I have updated the issue's title.

### As a...

student

### I can...

submit a recommendation request

### So that...

I can indicate my request to my professor

### Discussion

Student logs in and can access their profile
- Can see all their existing requests and their associated status
- Can create, read, update, delete, their recommendation requests

Through a Create form, a student can submit their request which:
- Has an autogenerated `id`
- Has a field for `professorName`
- Has a field for `professorEmail`
- Has a field for `requesterName`
- Has a dropdown that has `recommendationTypes`
- Also has comments section for `other` recommendation types (part of the dropdown, if selected the student can add more information in the details section)
- Has a section for student to add `details`
- Has `submissionDate` (set based on when the student submits the request)
- Has `completionDate` (set automatically based on when the professor completes the request)
- Has submit button

Recommendation Requests should also store `status` on the backend, this value needs to be set to Pending (awaiting professor review).

### Acceptance Criteria

- [ ] As a student, I can login and see all the requests I have submitted (but not any other students).
- [ ] There are endpoints for each of the CRUD actions necessary to create, read, update, and delete a recommendation request that have been tested on swagger.
- GET `api/recommendationrequest/{id}`
- PUT `api/recommendationrequest/{id}`
-  DELETE `api/recommendationrequest/{id}`
-  POST `api/recommendationrequest/post`

May also be useful to implement...
- GET by professorName `api/recommendationrequest/{professorName}`
- GET by studentName `api/recommendationrequest/{studentName}`

- [ ] A new recommendation request can be created from the create form (on the frontend) and is posted to the Recommendation Request Table.

### Implementation Todos

B1: Create Recommendation Request Table
- [x] Create entity and repository files for a Recommendation Request (`RecommendationRequest.java`, `RecommendationRequestRepository.java`)
- [x] There is a database migration file: `RecommendationRequest.json`
- [x] Table is visible on H2 console and visible when tested on Dokku

### PR DESCRIPTION
**Pull request by Mike-Petrus**
Closes #30 

Merge first, before PR # 39

H2 console on localhost
(will need to pull and run on local host if you test yourself)
20241115_15h26m25s_grim

Visible on dokku:
dokku postgres:connect rec-mike-petrus-dev-db
\dt

20241115_13h48m04s_grim

### CODE REVIEW

File: .github/workflows/82-kanban-slack-update.yml [hunk 1]
```diff
@@ -5,7 +5,7 @@ name: "82-kanban-slack-update.yml"
```
Comment (obenedek20) [inline .github/workflows/82-kanban-slack-update.yml]: Try and revert this file as it is not a part of your PR

File: src/main/java/edu/ucsb/cs156/rec/entities/RecommendationRequest.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+@Entity(name = "recommendationrequest")
+public class RecommendationRequest {
+  @Id
+  @GeneratedValue(strategy = GenerationType.IDENTITY)
+  private long id;
+
+  private String professorName;
+  private String professorEmail;
+  private String requesterName;
+  private String recommendationType;
```
Comment (obenedek20) [inline src/main/java/edu/ucsb/cs156/rec/entities/RecommendationRequest.java]: Epic specifies that this should be `recommendationTypes`

File: src/main/java/edu/ucsb/cs156/rec/entities/RecommendationRequest.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  private String details;
+
```
Comment (obenedek20) [inline src/main/java/edu/ucsb/cs156/rec/entities/RecommendationRequest.java:34]: missing `submissionDate`

File: src/main/java/edu/ucsb/cs156/rec/entities/RecommendationRequest.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  private LocalDateTime completionDate;
```
Comment (obenedek20) [inline src/main/java/edu/ucsb/cs156/rec/entities/RecommendationRequest.java:36]: Also missing a `status` (only shows up on backend)

File: src/main/resources/db/migration/changes/RecommendationRequest.json [hunk 5]
```diff
@@ -0,0 +1,80 @@
+{
+    "databaseChangeLog": [
```
Comment (obenedek20) [inline src/main/resources/db/migration/changes/RecommendationRequest.json:2]: Changes necessary for this page too based on changes to entities page

**Review by obenedek20 — CHANGES_REQUESTED** [review]

**Comment by Mike-Petrus** [comment]
Fixed above issues (status/date, kanban) and updated H2 screenshot

**Review by obenedek20 — APPROVED** [review]

**Review by pconrad — APPROVED** [review]
LGTM!

**Pull request merged**